src/index/FTSAlgorithms.cpp

```diff
@@ -290,14 +281,19 @@ Index::WordEntityPostings FTSAlgorithms::intersectKWay(
   if (entityMode) {
     resultWep.eids_.resize(n);
   }
+  for (size_t j = 0; j < k; j++) {
+    resultWep.wids_[j].resize(n);
+  }
```

Comment:
Because earlier we do a `resultWep.eids_.reserve(minSize)` and if pushed less than `minSize` elements we can free the extra space with the resize. (I know the name `minSize` is a bit misleading it isn't the minimum size of the result vector but the size of the smallest `wepVecs[i].cids_` vector)